### PR TITLE
drv/stm32h7-fmc-demo: ill-advised memory controller

### PR DESCRIPTION
**Pull request by cbiffle**
This is an exercise fixture for the PSRAM mode of the Flexible Memory Controller on the STM32H7, for evaluating FPGA interface options. It lets you do many ill-advised things via IPC. They all appear to work, no matter what the datasheet implies.

Good luck!

### CODE REVIEW

File: drv/stm32h7-fmc-demo-server/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    }
+
+    fn poke16(
+        &mut self,
+        _msg: &RecvMessage,
+        addr: u32,
+        value: u16,
+    ) -> Result<(), RequestError<Infallible>> {
+        let ptr = addr as *mut u16;
+        unsafe { ptr.write_volatile(value) }
```
Comment (hawkw) [inline drv/stm32h7-fmc-demo-server/src/main.rs:235]: hahahahha, _whew_ :D

File: drv/stm32h7-fmc-demo-server/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    }
+}
+
+impl idl::InOrderFmcDemoImpl for ServerImpl {
+    fn peek16(
+        &mut self,
+        _msg: &RecvMessage,
+        addr: u32,
+    ) -> Result<u16, RequestError<Infallible>> {
+        let ptr = addr as *const u16;
```
Comment (hawkw) [inline drv/stm32h7-fmc-demo-server/src/main.rs:213]: presumably pointer provenance would eventually prefer us to use [`ptr::from_exposed_addr`](https://doc.rust-lang.org/std/ptr/fn.from_exposed_addr.html) for ... all of these shenanigans, but that's not real yet...

**Review by hawkw — COMMENTED** [review]
this is deeply disquieting :)

**Review by nathanaelhuffman — APPROVED** [review]
lgtm, thanks for getting this set up!

**Pull request merged**